modules/services/picom.nix

```diff
@@ -0,0 +1,311 @@
+      description = ''
+        Opacity of active windows.
+      '';
+    };
+
+    inactiveDim = mkOption {
+      type = types.str;
+      default = "0.0";
+      example = "0.2";
+      description = ''
+        Dim inactive windows.
+      '';
+    };
+
+    inactiveOpacity = mkOption {
+      type = types.str;
+      default = "1.0";
+      example = "0.8";
+      description = ''
+        Opacity of inactive windows.
+      '';
+    };
+
+    menuOpacity = mkOption {
+      type = types.str;
+      default = "1.0";
+      example = "0.8";
+      description = ''
+        Opacity of dropdown and popup menu.
+      '';
+    };
+
+    opacityRule = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "87:class_i ?= 'scratchpad'" "91:class_i ?= 'xterm'" ];
+      description = ''
+        List of opacity rules.
+        See the
+        <citerefentry>
+          <refentrytitle>picom</refentrytitle>
+          <manvolnum>1</manvolnum>
+        </citerefentry>
+        man page for more examples.
+      '';
+    };
+
+    backend = mkOption {
+      type = types.str;
+      default = "glx";
+      description = ''
+        Backend to use: <literal>glx</literal> or <literal>xrender</literal>.
+      '';
+    };
+
+    vSync = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable vertical synchronization.
+      '';
+    };
+
+    refreshRate = mkOption {
+      type = types.int;
+      default = 0;
+      example = 60;
+      description = ''
+        Screen refresh rate (0 = automatically detect).
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.picom;
+      defaultText = literalExample "pkgs.picom";
+      example = literalExample "pkgs.picom";
+      description = ''
+        picom derivation to use.
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = types.str;
+      default = "";
+      example = ''
+        unredir-if-possible = true;
+        dbe = true;
+      '';
+      description = ''
+        Additional Picom configuration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    systemd.user.services.picom = {
+      Unit = {
+        Description = "Picom X11 compositor";
+        After = [ "graphical-session-pre.target" ];
+        PartOf = [ "graphical-session.target" ];
+      };
+
+      Install = { WantedBy = [ "graphical-session.target" ]; };
+
+      Service = let
+        experimentalBackendsFlag =
+          if cfg.experimentalBackends then " --experimental-backends" else "";
+      in {
+        ExecStart = "${cfg.package}/bin/picom --config ${configFile}"
+          + experimentalBackendsFlag;
+        Restart = "always";
+        RestartSec = 3;
+      } // optionalAttrs (cfg.backend == "glx") {
+        # Temporarily fixes corrupt colours with Mesa 18.
+        Environment = [ "allow_rgb10_configs=false" ];
+      };
```

Comment:
Do we still need this fix (introduced in https://github.com/rycee/home-manager/commit/9a0f388f66f6c7b388fecbbbb84cf0e95ef41fec), or was it fixed upstream already?

Reply:
Seems like the fix is still present in nixpkgs master at least, so I guess the answer is "yes":

https://github.com/NixOS/nixpkgs/blob/50281a823f220f74da8ea693955e60f7224cf4d9/nixos/modules/services/x11/picom.nix#L276-L279